### PR TITLE
feat: [CDS-98117]: Onboard provider APIS to terraform

### PR DESCRIPTION
**Pull request by sarthakkasat**
**Title:** [Component/Feature] Short Description of the Change

**Summary:**
Provide a brief overview of what the PR does and why it is needed.

**Details:**
Explain the changes in detail, including any relevant context or background information.

**Related Issues:**
Reference any related issues or tickets 

**Testing Instructions:**
Describe how the changes were tested. Include any test cases or steps to verify the functionality.
Please include the below test scenario with your changes.
- Create the resource and execute terraform apply. (Verify the resource is created)
- Execute terraform apply again without any changes. (Verify no changes should be done)
- Update the resource and execute terraform apply. (Verify the resource updated successfully)
- Remove the content from resource file and execute terraform apply. (Verify the resource has been deleted)
- Add again the resource and execute the terraform apply. (Verify the resource is created)
- Verify the import the resource file is working fine.
- In case of remote entity, Verify for both default and non default branch.

**Screenshots:**
Include before and after screenshots to visually demonstrate the changes.
**Checklist:**
- [x] Code changes are well-documented.
- [x] Tests have been added/updated.
- [x] Changes have been tested locally.
- [ ] Documentation has been updated.

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

File: internal/service/platform/provider/resource_provider.go [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+				Computed:    false,
+			},
+			"type": {
+				Description: "The type of the provider entity.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Required:    false,
+				Computed:    true,
+			},
+			"last_modified_at": {
```
Comment (mteodor) [inline internal/service/platform/provider/resource_provider.go:53]: is this field really configurable by the user?

File: internal/service/platform/provider/data_source_provider.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	resource := &schema.Resource{
+		Description: "Data source for Harness Provider.",
+
+		ReadContext: dataSourceProviderRead,
+
+		Schema: map[string]*schema.Schema{
+			"identifier": {
+				Description: "The identifier of the provider entity.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
```
Comment (mteodor) [inline internal/service/platform/provider/data_source_provider.go:22]: ForceNew shouldnt be used with data but with resource

File: internal/service/platform/provider/resource_provider.go (hunk 1 above)
Comment (sarthakkasat) [reply internal/service/platform/provider/resource_provider.go:53]: For this field, computed is marked as true

File: internal/service/platform/provider/data_source_provider.go (hunk 2 above)
Comment (sarthakkasat) [reply internal/service/platform/provider/data_source_provider.go]: removed force new

File: internal/service/platform/provider/resource_provider_test.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	"testing"
+
+	"github.com/harness/harness-go-sdk/harness/nextgen"
+	"github.com/harness/harness-go-sdk/harness/utils"
+	"github.com/harness/terraform-provider-harness/internal/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestResourceProvider(t *testing.T) {
```
Comment (lovish1999) [inline internal/service/platform/provider/resource_provider_test.go:14]: Can you add screenshots of tests run against either LOCAL or QA env ?

**Review by lovish1999 — APPROVED** [review]

File: internal/service/platform/provider/resource_provider_test.go (hunk 3 above)
Comment (sarthakkasat) [reply internal/service/platform/provider/resource_provider_test.go:14]: Done the testing using UTs

**Pull request merged**